### PR TITLE
feat(binder): Support bind args in Ordered-Set Agg

### PR DESCRIPTION
**Pull request by Honeta**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

as title

## Checklist For Contributors

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer to the issue)
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Checklist For Reviewers

- [ ] I have requested macro/micro-benchmarks as this PR can affect performance substantially, and the results are shown.

## Documentation

- [ ] My PR **DOES NOT** contain user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

File: src/frontend/src/binder/expr/function.rs [hunk 1]
```diff
@@ -154,12 +154,19 @@ impl Binder {
             .into());
         }
```
Comment (st1page) [inline src/frontend/src/binder/expr/function.rs:155]:
```suggestion
        } else {
        
        }
```
we need also check the Ordered-Set Agg Kinds followed with a within group clause

File: src/frontend/src/binder/expr/function.rs [hunk 2]
```diff
@@ -230,14 +237,53 @@ impl Binder {
...
-        let order_by = OrderBy::new(
-            f.order_by
+        let order_by = if f.within_group.is_some() {
+            OrderBy::new(
+                f.within_group
+                    .iter()
+                    .map(|x| self.bind_order_by_expr(*x.clone()))
+                    .try_collect()?,
+            )
```
Comment (st1page) [inline src/frontend/src/binder/expr/function.rs:246]: maybe we should return error if there is both within_group and order_by

File: src/frontend/src/binder/expr/function.rs [hunk 3]
```diff
@@ -154,12 +154,19 @@ impl Binder {
...
+                .iter()
+                .map(|x| self.bind_function_expr_arg(FunctionArgExpr::Expr(x.expr.clone())))
+                .flatten_ok()
+                .try_collect()?
+        } else {
+            f.args
+                .iter()
+                .map(|arg| self.bind_function_arg(arg.clone()))
+                .flatten_ok()
+                .try_collect()?
```
Comment (st1page) [inline src/frontend/src/binder/expr/function.rs:168]: should ignore the direct args here.

Comment (Honeta) [reply src/frontend/src/binder/expr/function.rs:176]: All direct args and other args is in `args` here.

Comment (Honeta) [reply src/frontend/src/binder/expr/function.rs:176]: The logic here is the same as before, does it need to be modified?

**Review by st1page — COMMENTED** [review]
please add some planner test, especially for those error branches

File: src/frontend/src/binder/expr/function.rs [hunk 4]
```diff
@@ -230,14 +243,61 @@ impl Binder {
...
+        } else {
+            OrderBy::new(
+                f.order_by
+                    .into_iter()
+                    .map(|e| self.bind_order_by_expr(e))
+                    .try_collect()?,
+            )
+        };
+        let direct_args = if matches!(kind, AggKind::PercentileCont | AggKind::PercentileDisc) {
+            let arg =
```
Comment (richardchien) [inline src/frontend/src/binder/expr/function.rs:269]: Although `arg` only contains one item, I suggest use `args` as its name.

**Review by richardchien — APPROVED** [review]
LGTM

**Pull request merged**